compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -640,6 +640,34 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // Extra info for `c"str"` before 2021 edition or `c "str"` in all editions. The heuristic
+        // here is to identify a cooked, uninterpolated `c` id immediately followed by a string, or
+        // a cooked, uninterpolated `cr` id immediately followed by a string or a `#`, in an
+        // edition where c-string literals are not allowed. There is the slight possibility of a
+        // false positive for a `cr#` that wasn't intended to start a c-string literal, but the
+        // lexer was greedy and didn't preserve whether the `r#` on its own would have started a
+        // valid raw string literal.
```

Comment:
This sentence is fine but you could also think about expanding upon it with something along the lines of “and we don't want to perform unbounded lookahead here to check if we have a sequence of hashes followed by a string literal”.